src/core/Akka.Cluster/Cluster.cs

```diff
@@ -80,10 +80,10 @@ public Cluster(ActorSystemImpl system)
             //create supervisor for daemons under path "/system/cluster"
             _clusterDaemons = system.SystemActorOf(Props.Create(() => new ClusterDaemon(_settings)).WithDeploy(Deploy.Local), "cluster");
 
-            //TODO: Pretty sure this is bad and will at least throw aggregateexception possibly worse. 
-            _clusterCore = GetClusterCoreRef().Result;
-
             _readView = new ClusterReadView(this);
+
+            // force the underlying system to warmup
+            GetClusterCoreRef();
```

Comment:
Blocking on `GetClusterCoreRef` in the constructor caused a deadlock reported by #1914, so we do two things now:

1. Call `GetClusterCoreRef` in the constructor to warm up the clustering system
2. Lazily acquire a reference to it as needed later, without blocking in the constructor.